ext4/tar2ext4/tar2ext4.go

```diff
@@ -237,12 +657,35 @@ func ReadExt4SuperBlock(vhdPath string) (*format.SuperBlock, error) {
 	return &sb, nil
 }
 
+func ReadExt4SuperBlockFromPartition(vhdPath string, partitionLBA int64) (*format.SuperBlock, error) {
+	partitionOffset := partitionLBA * compactext4.BlockSizeLogical
+	vhd, err := os.OpenFile(vhdPath, os.O_RDONLY, 0)
+	if err != nil {
+		return nil, err
+	}
+	defer vhd.Close()
+
+	// Skip padding at the start
+	if _, err := vhd.Seek(partitionOffset+1024, io.SeekStart); err != nil {
+		return nil, err
+	}
+	var sb format.SuperBlock
+	if err := binary.Read(vhd, binary.LittleEndian, &sb); err != nil {
+		return nil, err
+	}
+	// Make sure the magic bytes are correct.
+	if sb.Magic != format.SuperBlockMagic {
+		return nil, fmt.Errorf("not an ext4 file system, got %v", sb)
+	}
+	return &sb, nil
+}
+
 // ConvertAndComputeRootDigest writes a compact ext4 file system image that contains the files in the
 // input tar stream, computes the resulting file image's cryptographic hashes (merkle tree) and returns
 // merkle tree root digest. Convert is called with minimal options: ConvertWhiteout and MaximumDiskSize
 // set to dmverity.RecommendedVHDSizeGB.
 func ConvertAndComputeRootDigest(r io.Reader) (string, error) {
-	out, err := os.CreateTemp("", "")
+	out, err := ioutil.TempFile("", "")
```

Comment:
`ioutil` is deprecated as of Go 1.17. We should probably avoid using it. Was there a reason for this change?

Reply:
I'm not sure how this happened, might have been a bad rebase.